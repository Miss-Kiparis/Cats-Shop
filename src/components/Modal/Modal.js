import PropTypes from "prop-types";
import Button from "../Button/Button";
import "./Modal.scss";

function Modal(props) {
  return (
    <div
      data-testid="modal"
      className="overlay"
      onClick={(event) => {
        if (!event.target.closest(".modal")) {
          props.closeModal();
        }
      }}
    >
      <div className={props.modal__styles}>
        <div className="modal__content">
          <h2 className="modal__title">{props.title}</h2>
          <p className="modal__text">{props.text}</p>
          <div className="btn__container">
            <Button
              btnStyles={props.btn__styles + " btn__good"}
              text="Add to cart"
              btnAction={() => {
                props.closeModal();
                props.cartListHandler(props.activeCart);
              }}
            />
            <Button
              btnStyles={props.btn__styles + " btn__bad"}
              text="Cancel"
              btnAction={props.closeModal}
            />
          </div>

          {props.closeModalBtn && (
            <button onClick={props.closeModal} className="modal__close">
              X
            </button>
          )}
        </div>
      </div>
    </div>
  );
}

export default Modal;

Modal.defaultProps = {
  btn__styles: "btn__styles",
  modal__styles: "modal__styles",
  closeModalBtn: true,
};

Modal.propTypes = {
  title: PropTypes.string,
  text: PropTypes.string,
  closeModal: PropTypes.func,
  handlerModal: PropTypes.func,
  cartListHandler: PropTypes.func,
  activeCart: PropTypes.string,
};
