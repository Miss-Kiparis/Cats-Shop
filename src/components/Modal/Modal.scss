.overlay{
    background-color: rgb(190 190 190 / 73%);
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    height: 100%;
    width: 100%;
    z-index: 1;
}

.modal__text {
    color: #000;
    padding: 15px;
    font-size: 22px;
    text-align: center;
}

.modal {
    width: 650px;
    margin: 200px auto auto;
    position: relative;

 
}

.modal__content {
    border-radius: 30px;
    background: #e0e0e0;
    box-shadow: 15px 15px 30px #bebebe,
               -15px -15px 30px #ffffff;

}

.modal__close {
    padding: 10px 15px;
    position: relative;
    bottom: 200px;
    left: 595px;
    background-color: #ffffff;
    border: solid #fff 1px;
    border-radius: 15px;
    color: #000;
    outline: none;
    cursor: pointer;
    transition: all 0.4s;}

.modal__close:hover {
    background-color: #dad9d9;
    border: solid #dad9d9 1px;
    box-shadow: 
    inset 0 0 0 10px #e4341a;
    color: #e4341a;

}


.modal__title {
    background-color: #5a4f4f85;
    margin: 0;
    color: #fff;
    padding: 15px;
    border-top-left-radius: 30px;
    border-top-right-radius: 30px;


}


.modal__text {
    color: #000;
    padding: 15px;
    text-align: center;
}