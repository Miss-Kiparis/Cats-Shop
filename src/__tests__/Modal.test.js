import "@testing-library/jest-dom";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import Modal from "../components/Modal/Modal";
import renderer from "react-test-renderer";

describe("Modal", () => {
  afterEach(() => {
    cleanup();
  });

  const mockData = {
    cartListHandler: jest.fn(),
    modal__styles: "mock_style",
    btn__styles: " ",
    closeModal: jest.fn(),
    closeModalBtn: true,
    title: "mock title",
    text: "mock text",
    activeCart: "mock id",
  };

  test("should exist", () => {
    render(
      <Modal
        cartListHandler={mockData.cartListHandler}
        modal__styles={mockData.modal}
        btn__styles={mockData.btn__styles}
        closeModal={mockData.closeModal}
        closeModalBtn={mockData.closeModalBtn}
        title={mockData.title}
        text={mockData.text}
        activeCart={mockData.activeCart}
      />
    );
    const modal = screen.getByTestId("modal");
    expect(modal).toBeInTheDocument();
  });

  test("should have some text", () => {
    render(
      <Modal
        cartListHandler={mockData.cartListHandler}
        modal__styles={mockData.modal}
        btn__styles={mockData.btn__styles}
        closeModal={mockData.closeModal}
        closeModalBtn={mockData.closeModalBtn}
        title={mockData.title}
        text={mockData.text}
        activeCart={mockData.activeCart}
      />
    );
    const modal = screen.getByTestId("modal");
    expect(modal).toHaveTextContent("mock text");
  });

  test("should have cartListHandler", () => {
    render(
      <Modal
        cartListHandler={mockData.cartListHandler}
        modal__styles={mockData.modal}
        btn__styles={mockData.btn__styles}
        closeModal={mockData.closeModal}
        closeModalBtn={mockData.closeModalBtn}
        title={mockData.title}
        text={mockData.text}
        activeCart={mockData.activeCart}
      />
    );
    const goodButton = screen.getByText("Add to cart");
    fireEvent.click(goodButton);
    expect(mockData.cartListHandler).toHaveBeenCalled();
  });

  test("should have closeModal", () => {
    render(
      <Modal
        cartListHandler={mockData.cartListHandler}
        modal__styles={mockData.modal}
        btn__styles={mockData.btn__styles}
        closeModal={mockData.closeModal}
        closeModalBtn={mockData.closeModalBtn}
        title={mockData.title}
        text={mockData.text}
        activeCart={mockData.activeCart}
      />
    );
    const modal = screen.getByTestId("modal");
    fireEvent.click(modal);
    expect(mockData.closeModal).toHaveBeenCalled();
  });
});

test("matches snapshot", () => {
  const mockData = {
    cartListHandler: jest.fn(),
    modal__styles: "mock_style",
    btn__styles: " ",
    closeModal: jest.fn(),
    closeModalBtn: true,
    title: "mock title",
    text: "mock text",
    activeCart: "mock id",
  };
  const tree = renderer
    .create(
      <Modal
        cartListHandler={mockData.cartListHandler}
        modal__styles={mockData.modal}
        btn__styles={mockData.btn__styles}
        closeModal={mockData.closeModal}
        closeModalBtn={mockData.closeModalBtn}
        title={mockData.title}
        text={mockData.text}
        activeCart={mockData.activeCart}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
