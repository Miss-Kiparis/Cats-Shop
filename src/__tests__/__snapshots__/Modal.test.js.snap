// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="overlay"
  data-testid="modal"
  onClick={[Function]}
>
  <div
    className="modal__styles"
  >
    <div
      className="modal__content"
    >
      <h2
        className="modal__title"
      >
        mock title
      </h2>
      <p
        className="modal__text"
      >
        mock text
      </p>
      <div
        className="btn__container"
      >
        <button
          className="  btn__good"
          data-testid="btn"
          onClick={[Function]}
          type="button"
        >
          Add to cart
        </button>
        <button
          className="  btn__bad"
          data-testid="btn"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
      </div>
      <button
        className="modal__close"
        onClick={[MockFunction]}
      >
        X
      </button>
    </div>
  </div>
</div>
`;
